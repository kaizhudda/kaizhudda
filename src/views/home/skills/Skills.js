import React from 'react';
import Title from '../../../sharedComponents/title';
import Skill from './skill';

import './Skills.scss';

const Skills = () => {
  return (
    <div className="Skills row center-xs" id="skills-section">
      <Title>Skills</Title>
      <Skill skills={feSkills} category="Front-End" />
      <Skill skills={beSkills} category="Back-End" />
      <Skill skills={otherSkills} category="CI/CD and other" />
    </div>
  );
}

export default Skills;

const feSkills = [
  {
    name: "React",
    experience: "3+ years"
  },
  {
    name: "Redux",
    experience: "3+ years"
  }, 
  {
    name: "JavaScript/ES6",
    experience: "6+ years"
  }, 
  {
    name: "HTML5",
    experience: "10+ years"
  }, 
  {
    name: "CSS3",
    experience: "8+ years"
  }, 
  {
    name: "SASS",
    experience: "3+ years"
  },
  {
    name: "Jest/Enzyme",
    experience: "2+ years"
  }
];

const beSkills = [
  {
    name: "Node.js",
    experience: "3+ years"
  },
  {
    name: "C#/.NET",
    experience: "2+ years"
  }, 
  {
    name: "Java",
    experience: "1 year"
  }, 
  {
    name: "SQL Server",
    experience: "2+ years"
  }, 
  {
    name: "MongoDB",
    experience: "1+ years"
  }, 
  {
    name: "Express.js",
    experience: "1+ years"
  }
];

const otherSkills = [
  {
    name: "Microsoft Azure",
    experience: "2+ years"
  },
  {
    name: "Docker",
    experience: "1+ years"
  }, 
  {
    name: "Jenkins",
    experience: "1 year"
  },
  {
    name: "Git",
    experience: "6+ years"
  }
];
