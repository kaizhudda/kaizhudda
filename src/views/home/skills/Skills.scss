@import '../../../styles/variables';
@import '../../../styles/media-queries';

.Skills {
  background: white;
  color: #727878;
  font-family: $font-body;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 16px 16px 48px 16px;

  .skills-content {
    display: flex;
    flex-direction: column;
    width: 550px;

    .skill {
      width: 100%;
      display: flex;
      justify-content: space-between;
      border: 1px solid #363636;
      margin-bottom: 10px;
      &:hover {
        background: #efefef;

        .fa-star {
          font-size: 1.125rem;
        }

        .fa-star-half-alt {
          font-size: 1.125rem;
        }
      }

      .skill-title {
        width: 70%;
        padding: 10px;
        text-align: left;
      }

      .skill-score {
        width: 30%;
        padding: 10px;
        text-align: right;

        .fa-star {
          transition: all 0.2s cubic-bezier(.25,.8,.25,1);
          color: #22a39f;
        }

        .fa-star-half-alt {
          transition: all 0.2s cubic-bezier(.25,.8,.25,1);
          color: #22a39f;
        }

        

      }
    }
  }
}