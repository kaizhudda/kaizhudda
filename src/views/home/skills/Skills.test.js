import React from 'react';
import { shallow } from 'enzyme';
import Skills from './Skills';

let props = {};
let wrapped = shallow(<Skills {...props} />); 

describe('Skills', () => {
  it('should render the Skills Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should render skill components', () => {
    expect(wrapped.find('Skill').length).toBe(3);
  });
});
