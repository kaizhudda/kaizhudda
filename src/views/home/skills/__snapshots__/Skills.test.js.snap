// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills should render the Skills Component correctly 1`] = `
<div
  className="Skills row center-xs"
  id="skills-section"
>
  <Title>
    Skills
  </Title>
  <Skill
    category="Front-End"
    skills={
      Array [
        Object {
          "experience": "3+ years",
          "name": "React",
        },
        Object {
          "experience": "3+ years",
          "name": "Redux",
        },
        Object {
          "experience": "5+ years",
          "name": "JavaScript/ES6",
        },
        Object {
          "experience": "5+ years",
          "name": "HTML5",
        },
        Object {
          "experience": "5+ years",
          "name": "CSS3",
        },
        Object {
          "experience": "3+ years",
          "name": "SASS",
        },
        Object {
          "experience": "3+ years",
          "name": "Jest/Enzyme",
        },
      ]
    }
  />
  <Skill
    category="Back-End"
    skills={
      Array [
        Object {
          "experience": "2+ years",
          "name": "Node.js",
        },
        Object {
          "experience": "2+ years",
          "name": "C#/.NET",
        },
        Object {
          "experience": "1 year",
          "name": "Java",
        },
        Object {
          "experience": "2+ years",
          "name": "SQL Server",
        },
        Object {
          "experience": "1+ years",
          "name": "MongoDB",
        },
        Object {
          "experience": "1+ years",
          "name": "Express.js",
        },
      ]
    }
  />
  <Skill
    category="CI/CD and other"
    skills={
      Array [
        Object {
          "experience": "2+ years",
          "name": "Microsoft Azure",
        },
        Object {
          "experience": "1+ years",
          "name": "Docker",
        },
        Object {
          "experience": "1 year",
          "name": "Jenkins",
        },
        Object {
          "experience": "5+ years",
          "name": "Git",
        },
      ]
    }
  />
</div>
`;
